File: Problem 14/C++/problem14.cpp
#include <iostream>
#include <time.h>

unsigned int collatz(unsigned long int k)
{
	int i = 0;
	while(k != 1)
	{
		if(k % 2 == 0)
			k = k / 2;
		else
			k = 3*k + 1;
		++i;
	}
	return i;
}

int main()
{
	time_t t1;
	time(&t1);
	int savedNum = 0;
	int savedIter = 0;
	for(int i = 2; i < 1000000; ++i)
	{
		int k = collatz(i);
		if (k > savedIter)
		{
			savedIter = k;
			savedNum = i;
		}
	}
	time_t t2;
	time(&t2);
	std::cout << "(" << savedNum <<", " << savedIter << ")" << std::endl;
	std::cout << "Runtime was " <<difftime(t2,t1) << std::endl;
	return 0;
}

File: Problem 14/Python/problem14.py
import time
def Collatz(k):
	i = 0
	while k != 1:
		if k % 2 == 0:
			k = k / 2
		else:
			k = 3*k + 1
		i+=1
	return i


def main():
	t1 = time.time()
	tup = (0,0)
	i = 2
	while i < 1000000:
		k = Collatz(i)
		if k > tup[1]:
			tup = (i,k)
		i += 1
	print(tup, time.time() - t1)

if __name__ == '__main__':
	main()